cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/Capacity.java

```diff
@@ -178,8 +229,109 @@ public static Double getSizeInMiB(String size) {
      * @param throughput The String representation of the throughput.
      * @return The equivalent number of kibibytes.
      */
-    public static Double getThroughputInKiB(String throughput) {
+    public static String getThroughputInKiB(String throughput) {
         String size = throughput.substring(0, throughput.indexOf("B"));
-        return StorageUtils.convertTo(size, "Ki");
+        return String.valueOf(StorageUtils.convertTo(size, "Ki"));
+    }
+
+    private void processCapacityEntries(BrokerCapacity bc, Storage s) {
+        String cpu = processCpu();
+        String disk = processDisk(storage, Broker.DEFAULT_BROKER_ID);
+        String inboundNetwork = processInboundNetwork(bc, null);
+        String outboundNetwork = processOutboundNetwork(bc, null);
+
+        // Default broker entry
+        Broker defaultBroker = new Broker(Broker.DEFAULT_BROKER_ID, cpu, disk, inboundNetwork, outboundNetwork);
+        capacityEntries.put(Broker.DEFAULT_BROKER_ID, defaultBroker);
+
+        if (storage instanceof JbodStorage) {
+            // A capacity configuration for a cluster with a JBOD configuration
+            // requires a distinct broker capacity entry for every broker because the
+            // Kafka volume paths are not homogeneous across brokers and include
+            // the broker pod index in their names.
+            for (int id = 0; id < replicas; id++) {
+                disk = processDisk(storage, id);
+                Broker broker = new Broker(id, cpu, disk, inboundNetwork, outboundNetwork);
+                capacityEntries.put(id, broker);
+            }
+        }
+
+        if (bc != null) {
+            // For checking for duplicate brokerIds
+            HashSet<Integer> overrideIds = new HashSet<>();
+            List<BrokerCapacityOverride> overrides = bc.getOverrides();
+            // Override broker entries
+            if (overrides != null) {
+                if (overrides.isEmpty()) {
+                    LOGGER.warnOp("Ignoring empty overrides list");
+                } else {
+                    for (BrokerCapacityOverride override : overrides) {
+                        List<Integer> ids = override.getBrokers();
+                        inboundNetwork = processInboundNetwork(bc, override);
+                        outboundNetwork = processOutboundNetwork(bc, override);
+                        for (Integer id : ids) {
+                            if (id == Broker.DEFAULT_BROKER_ID) {
```

Comment:
This works because `Broker.DEFAULT_BROKER_ID` is an `int` and causes `id` to auto unbox. It's usually preferable to avoid relying on this behavior in equality checks.